assets/js/Ioda/pages/dashboard/DashboardTab.js

```diff
@@ -16,62 +16,63 @@ import { asn } from "./DashboardConstants";
 import { AreaChartOutlined, GlobalOutlined } from "@ant-design/icons";
 import { getSecondsAsErrorDurationString } from "../../utils/timeUtils";
 
-class DashboardTab extends Component {
-  constructor(props) {
-    super(props);
-    this.config = React.createRef();
+const DashboardTab = (props) => {
+  console.log(props);
 
-    this.state = {
-      mounted: false,
-    };
-  }
+  const {
+    eventDataProcessed,
+    topoData,
+    topoScores,
+    handleEntityShapeClick,
+    activeTabType,
+    until,
+    from,
+    totalOutages,
+    type,
+    tabCurrentView,
+    handleTabChangeViewButton,
+    summaryDataRaw,
+    genSummaryTableDataProcessed,
+    summaryDataProcessed
+  } = props;
 
-  componentDidMount() {
-    this.setState({ mounted: true });
-  }
 
-  componentWillUnmount() {
-    this.setState({ mounted: false });
-  }
+    const config = React.createRef();
+    const [mounted, setMounted] = useState(false);
 
-  componentDidUpdate(prevProps) {
-    if (!this.state.mounted) {
-      return;
-    }
+  useEffect(() => {
+    setMounted(true);
+  }, []);
 
-    if (
-      this.props.eventDataProcessed !== prevProps.eventDataProcessed &&
-      this.config.current
-    ) {
-      this.genChart();
+  useEffect(() => {
+    if(eventDataProcessed != null && config.current) {
+      genChart();
     }
-  }
+  }, [eventDataProcessed]);
 
-  genMap = () => {
+  function genMap () {
     return (
       <TopoMap
-        topoData={this.props.topoData}
-        scores={this.props.topoScores}
-        handleEntityShapeClick={(entity) =>
-          this.props.handleEntityShapeClick(entity)
-        }
-        entityType={this.props.activeTabType?.toLowerCase()}
+        topoData={topoData}
+        scores={topoScores}
+        handleEntityShapeClick={(entity) => handleEntityShapeClick(entity)}
+        entityType={activeTabType?.toLowerCase()}
       />
     );
-  };
+  }
```

Comment:
I don't see a reason to have this as a separate function. We should be able to just include the <TopoMap> component in the render JSX directly, and get rid of this altogether.

I know this wasn't something you introduced, just a suggestion to improve the codebase.